### ISSUE TITLE
SigLIP learned posemb has extreme L2 norm disparity across the patch grid, when loading openpi's official pi05 ckpt

### ISSUE THREAD

**Issue by xiaojxkevin** [Physical-Intelligence/openpi#947]
### Observation

When visualizing action→image attention heatmaps on `pi0` / `pi0.5`, certain image locations consistently show high attention across layers, heads, and inputs regardless of image content. As shown in the figure below:

<img width="224" height="248" alt="Image" src="https://github.com/user-attachments/assets/aae3600c-63d8-4935-8814-0522bdc12c3f" />

### Finding

The SigLIP vision encoder (So400m/14, width=1152, patch grid 16×16) uses a learned positional embedding (`posemb="learn"`). Analysis of the **`pi05_base` checkpoint (from openpi's official checkpoint)** reveals extreme L2 norm disparity across the 16×16 patch grid:

<img width="320" height="320" alt="Image" src="https://github.com/user-attachments/assets/6f138d04-c1d3-4af9-9684-b9d19b92d266" />

Key numbers:

| Patch (row, col) | L2 Norm | ×mean |
|---|---|---|
| (12, 1) | **313.25** | 14.5× |
| (2, 15) | **231.12** | 10.7× |
| (3, 0) | **197.95** | 9.2× |
| (0, 13) | 93.84 | 4.4× |
| (15, 3) | 97.34 | 4.5× |

- Interior patches (196 patches): L2 norm ≈ **17–19** (very uniform)
- Anomalous edge patches: L2 norm **33–313** (up to **15×** mean)
- One dimension (`dim 1076`) contributes **82%** of the total norm for patch (12, 1), with value +283.4 vs −0.34 at center — consistent with an **attention sink** pattern

### Mechanism

```python
# siglip.py:336 — posemb is added BEFORE the SigLIP transformer
x = x + get_posemb(self, self.posemb, (h, w), c, "pos_embedding", jnp.float32)
```

The posemb is a learned parameter (shape `[1, 256, 1152]`, initialized with `normal(stddev=1/sqrt(1152))`). During pretraining, certain edge positions developed extreme magnitudes, producing a content-independent contribution to `q @ k^T` for those positions.

The SigLIP head projects 1152 → 2048 (PaliGemma width), so the posemb magnitude anomaly propagates into the LLM attention layers.

- `src/openpi/models/siglip.py` — SigLIP config with `posemb="learn"`
- `src/openpi/models/model.py` — `preprocess_observation`

### Reproduction

The visualization script is given: [viz_siglip_posemb.py](https://github.com/user-attachments/files/27708943/viz_siglip_posemb.py)
```bash
uv run python tools/viz_siglip_posemb.py --save_dir ./siglip
```

**Comment by xiaojxkevin** [comment]
Some updates: these maybe the cases of ViT registers, and this cannot be simply regarded as "wrong".
